src/code/cutscene_camera.c

```diff
@@ -353,28 +357,579 @@ void CutsceneCamera_Reset(void) {
+                                              CsCmdCamMisc* miscCmd, CutsceneCameraInterp* interpState) {
+    f32 new_var;
+    f32 coeff[3];
+    s32 sp3C[3];
+
+    if (interpState->type != CS_CAM_INTERP_MP_QUAD) {
+        // Initialize
+        interpState->type = CS_CAM_INTERP_MP_QUAD;
+        interpState->waypoint = 0;
+        interpState->duration = pointCmd->duration;
+        interpState->curFrame = 0;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/cutscene_camera/func_80163334.s")
+    new_var = (f32)interpState->curFrame / pointCmd[interpState->waypoint + 1].duration;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/cutscene_camera/func_80163660.s")
+    if (interpState->waypoint < (interpState->duration - 1)) {
+        sp3C[0] = interpState->waypoint;
+    } else {
+        sp3C[0] = interpState->duration - 1;
+    }
+
+    if ((interpState->waypoint + 1) < (interpState->duration - 1)) {
+        sp3C[1] = interpState->waypoint + 1;
+    } else {
+        sp3C[1] = interpState->duration - 1;
+    }
+
+    if ((interpState->waypoint + 2) < (interpState->duration - 1)) {
+        sp3C[2] = interpState->waypoint + 2;
+    } else {
+        sp3C[2] = interpState->duration - 1;
+    }
+
+    func_801624EC(new_var, coeff);
+
+    if (camPos != NULL) {
+        camPos->x = (coeff[0] * pointCmd[sp3C[0]].pos.x) + (coeff[1] * pointCmd[sp3C[1]].pos.x) +
+                    (coeff[2] * pointCmd[sp3C[2]].pos.x);
+        camPos->y = (coeff[0] * pointCmd[sp3C[0]].pos.y) + (coeff[1] * pointCmd[sp3C[1]].pos.y) +
+                    (coeff[2] * pointCmd[sp3C[2]].pos.y);
+        camPos->z = (coeff[0] * pointCmd[sp3C[0]].pos.z) + (coeff[1] * pointCmd[sp3C[1]].pos.z) +
+                    (coeff[2] * pointCmd[sp3C[2]].pos.z);
+    }
+
+    if (camFov != NULL) {
+        *camFov =
+            (coeff[0] * miscCmd[sp3C[0]].fov) + (coeff[1] * miscCmd[sp3C[1]].fov) + (coeff[2] * miscCmd[sp3C[2]].fov);
+    }
+
+    if (camRoll != NULL) {
+        s16 sp34[3];
+        s32 sp28[2];
+        s32 temp;
+
+        sp34[0] = CAM_DEG_TO_BINANG(miscCmd[sp3C[0]].roll);
+        sp34[1] = CAM_DEG_TO_BINANG(miscCmd[sp3C[1]].roll);
+        sp34[2] = CAM_DEG_TO_BINANG(miscCmd[sp3C[2]].roll);
+
+        sp28[0] = (s16)(sp34[1] - sp34[0]);
+        sp28[1] = sp28[0] + (s16)(sp34[2] - sp34[1]);
+
+        temp = ((coeff[1] * sp28[0]) + (coeff[2] * sp28[1]));
+        *camRoll = sp34[0] + temp;
+    }
+
+    interpState->curFrame++;
+
+    if (interpState->curFrame == pointCmd[interpState->waypoint + 1].duration) {
+        interpState->waypoint++;
+        interpState->curFrame = 0;
+        if (interpState->waypoint >= (interpState->duration - 2)) {
+            interpState->type = CS_CAM_INTERP_OFF;
+            return interpState->duration;
+        }
+    }
+    return 0;
+}
+
+/**
+ * This code is very similar to the unused spline system in OoT's func_800BB0A0
+ * in which that is based on the Super Mario 64 cutscene camera movement
```

Comment:
This wording here on the second line seems a bit off? How about:
```suggestion
 * This code is very similar to the unused spline system in OoT's func_800BB0A0
 * in that it is based on the Super Mario 64 cutscene camera movement
```